shared/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -119,12 +120,47 @@ namespace datadog::shared::nativeloader
         const auto process_name = ::shared::GetCurrentProcessName();
         Log::Debug("ProcessName: ", process_name);
 
-        if (process_name == WStr("dd-trace") || process_name == WStr("dd-trace.exe"))
+        Log::Debug("CorProfiler::Initialize");
+
+        const auto& include_process_names = GetEnvironmentValues(EnvironmentVariables::IncludeProcessNames);
+
+        // if there is a process inclusion list, attach profiler only if this
+        // process's name is on the list
+        if (!include_process_names.empty() && !Contains(include_process_names, process_name))
         {
-            Log::Info("Profiler disabled - monitoring the dd-trace tool is not supported.");
+            Log::Info("CorProfiler::Initialize Profiler disabled: ", process_name, " not found in ",
+                         EnvironmentVariables::IncludeProcessNames, ".");
             return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
         }
 
+        // if we were on the explicit include list, don't check the block list
+        if (include_process_names.empty())
+        {
+            const auto& exclude_process_names = GetEnvironmentValues(EnvironmentVariables::ExcludeProcessNames);
+            if (!exclude_process_names.empty())
+            {
+                // attach profiler only if this process's name is NOT on the provided list
+                if (Contains(exclude_process_names, process_name))
+                {
+                    Log::Info("CorProfiler::Initialize Profiler disabled: ", process_name, " found in ",
+                                 EnvironmentVariables::ExcludeProcessNames, ".");
+                    return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
+                }
+            }
+            else
+            {
+                // attach profiler only if this process's name is NOT on the default block list
+                for (auto&& exclude_assembly : default_exclude_assemblies)
```

Comment:
Hmmm. Yeah, that probably makes sense🤔 That way customers/we can easily _extend_ the default list rather than having to _replace_ them, so less likely to get escalations about people accidentally removing them. And they can always add to the include list if necessary. I like it 👍